### DISCUSSION TITLE
Is it possible to mix on demand ISR and SSR on the same route segment?

### DISCUSSION THREAD

**Discussion by KraKeN-47** [vercel/next.js#73762] [App Router]
Hello,

I have a case where there are 2 different content pages on the same route segment. However, I know which specific routes can be statically generated, however - the rest must be SSR'ed. The question I have, is it possible to mix SSR and On-demand ISR? The specified pages could be On demand ISR'ed, while non specified ones would be always SSR'ed

**Comment by icyJoseph** [comment]
That was a bit confusion. SSR is everything, I guess you mean, pre-render up to some point, and don't pre-render the rest? 

I think this is what PPR aims to do. There are some interesting developments coming, for example: https://nextjs.org/docs/app/api-reference/functions/connection

**Reply by KraKeN-47** [reply]
This is exactly what I mean. I have a list of pages that I know can be ISR'ed, the rest - can't. However, they fall under the same route segment [[...slug]] ( not a mistake ), ppr is not an option as it is not stable and prod ready yet

**Reply by icyJoseph** [reply]
Uhhh that's a bit more complex I'd say. I suppose having the connection function could be the way forward.
